### PR TITLE
auto detect browser

### PR DESCRIPTION
**Pull request by mrnninster**
This PR resolves the #9 and #10 issues
It checks for a valid browser within the directory if the browser path flag is not set, so it is optional
returns an error if the flag is not set and there are no valid browsers(#9 )

The debug flag also provides instruction on adding the browser path.

### CODE REVIEW

**Review by caffo — CHANGES_REQUESTED** [review]
Please remove the downloaded files (formatted, json and markdown) from the PR.

**Comment by caffo** [comment]
Tests are now failing on this branch :)

**Comment by mrnninster** [comment]
> Please remove the downloaded files (formatted, json and markdown) from the PR.


I have removed them


> Tests are now failing on this branch :)

Yes they are failing with a permission denied

```bash
./tests.py
  shell: /usr/bin/bash -e {0}
  env:
    OS: ubuntu-latest
    PYTHON: [3](https://github.com/caffo/roam-to-git/actions/runs/4856066952/jobs/8655235071?pr=16#step:7:3).8
    pythonLocation: /opt/hostedtoolcache/Python/3.8.16/x6[4](https://github.com/caffo/roam-to-git/actions/runs/4856066952/jobs/8655235071?pr=16#step:7:4)
    LD_LIBRARY_PATH: /opt/hostedtoolcache/Python/3.8.16/x64/lib
/home/runner/work/_temp/dcd9c363-b209-41eb-90b[5](https://github.com/caffo/roam-to-git/actions/runs/4856066952/jobs/8655235071?pr=16#step:7:5)-9cba3f55f[6](https://github.com/caffo/roam-to-git/actions/runs/4856066952/jobs/8655235071?pr=16#step:7:6)3[7](https://github.com/caffo/roam-to-git/actions/runs/4856066952/jobs/8655235071?pr=16#step:7:7).sh: line 1: ./tests.py: Permission denied
```

I checked the test that passed and this and they are both the same
I am not sure what permissions are getting denied

File: tests.py
Comment (caffo) [inline tests.py:1]: Is it empty? Compare it with the master branch.

**Review by caffo — COMMENTED** [review]
Question on tests.py

File: tests.py
Comment (mrnninster) [reply tests.py:1]: The test are now passing
The _"test.py"_ file wasn't detected as an executable that was the reason for the permission error

fixed that by doing this
```bash
git update-index --chmod=+x ./test.py
```

**Review by caffo — APPROVED** [review]

**Pull request merged**